blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/PipelineSearch.java

```diff
@@ -107,11 +108,21 @@ public String getType() {
             });
         }else{
             GlobMatcher matcher = pipeline.contains("*") ? new GlobMatcher(pipeline) : null;
-            while (pipelineIterator.hasNext()) {
-                BluePipeline p = pipelineIterator.next();
-                // If using glob syntax try to match using the glob matcher otherwise fall back to equality check
-                if (matcher != null && matcher.matches(p.getFullName())) {
-                    pipelines.add(p);
+            if (matcher != null) {
+                while (pipelineIterator.hasNext()) {
+                    BluePipeline p = pipelineIterator.next();
+                    String decodedName = URLDecoder.decode(p.getFullDisplayName());
```

Comment:
Fixed the deprecated API use.

I'm using `getFullDisplayName` as we are filtering based on what the user sees. Perhaps this should be moved to a different omnisearch implementation that is just used for user search?